### PR TITLE
os/dbuild: Add download option for selecting specific board

### PR DESCRIPTION
**Pull request by tyha1115**
Previously, when user connected serveral boards to download the binary, the binary was able to be downloaded only to the last connected board.
Now, a user can select the specific board for downloading binary. (But only support Arkik05x boards)
If a user want to select the specific board, please use './dbulid.sh menu' to download the binary.

After your build was finished,
```sh
======================================================
  "Select build Option"
======================================================
  "1. Build with Current Configurations"
  "2. Re-configure"
  "3. Menuconfig"
  "4. Build Clean"
  "5. Build Dist-Clean"
  "d. Download"
  "x. Exit"
======================================================
```

After select 'Download' option, you should select the board type.
(If you have already selected a board when you run the script this time, this process is skipped.)
```sh
======================================================
  "Select Board"
======================================================
  "1. artik053"
  "2. artik053s"
  "3. artik055s"
  "4. cy4390x"
  "5. qemu"
  "6. sidk_s5jt200"
  "x. EXIT"
======================================================
```
If user selects 'artik05x' boards, user can choose PORT Select Option.
```sh
=============================================================================
  "Select PORT Option"
=============================================================================
  "1. Last Connected Device"
  "2. Select Specific Port"
  "x. Exit"
=============================================================================
```

If user want to select the specific board, the script gives BUS and PORT information, and user can select the specific BUS and PORT.
```sh
=============================================================================
  "BUS and PORT Information of Available Device"
=============================================================================
/:  Bus 02: Dev 1, Class=root_hub, Driver=uhci_hcd/2p, 12M
/:  Bus 01: Dev 1, Class=root_hub, Driver=ehci-pci/6p, 480M
    |__ Port 1: Dev 3, Class=Vendor Specific Class, Driver=ftdi_sio, 480M
    |__ Port 2: Dev 4, Class=Vendor Specific Class, Driver=ftdi_sio, 480M
=============================================================================
  "Select BUS and PORT"
=============================================================================
  "1. Bus: 1 Port: 1"
  "2. Bus: 1 Port: 2"
  "x. EXIT"
=============================================================================
```

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [35378a5742f96b74aa337e2154e9740087b0e10e] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [35378a5742f96b74aa337e2154e9740087b0e10e] - Code Rule Check OK.

File: os/dbuild.sh [hunk 1]
```diff
@@ -247,6 +250,103 @@ function SELECT_CONFIG()
...
+{
+	# Get current target board information
+	if [ -z ${BOARD} ]; then
+		SELECT_BOARD
+	fi
+
+	case ${BOARD} in
+	artik053|artik053s|artik055s)
+		unset SELECTED_PORT
+		CFG_FILE=${CONFIGDIR}/artik05x/scripts/artik05x-usb-ftdi.cfg
```
Comment (jeongchanKim) [inline os/dbuild.sh]: This path can be changed when file name or path is changed.
How about moving this to the above?

File: os/dbuild.sh [hunk 2]
```diff
@@ -247,6 +250,103 @@ function SELECT_CONFIG()
...
+		SELECT_DL
+		;;
+	2|select*)
+		# Get the number list of connected artik05x devices
+		DEVICE_LIST=`lsusb -d 0403:6010 | awk '{printf "%d ", NR}'`
+
+		# Get buses which are connected with artik05x
+		BUS_NUMS=`lsusb -d 0403:6010 | awk '{printf "%d ", $2}'`
+
+		# Remove duplcated bus numbers and sort
```
Comment (jeongchanKim) [inline os/dbuild.sh]: duplcated -> duplicated

File: os/dbuild.sh [hunk 3]
```diff
@@ -247,6 +250,103 @@ function SELECT_CONFIG()
...
+				echo "  \"${IDX}. Bus: ${BUS_IDX} Port: ${BUS_PORT_IDX}\""
+				BUS_PORT[${IDX}]="${BUS_IDX}:${BUS_PORT_IDX}"
+				((IDX=IDX+1))
+			done
+		done
+		echo "  \"x. EXIT\""
+		echo ==================================================================================
+		read SELECTED_BUS_PORT
+
+		# treate "exit"
```
Comment (jeongchanKim) [inline os/dbuild.sh]: treate -> treat

File: os/dbuild.sh [hunk 4]
```diff
@@ -247,6 +250,103 @@ function SELECT_CONFIG()
...
+		if [ "${SELECTED_BUS_PORT}" == "x" -o "${SELECTED_BUS_PORT}" == "exit" -o "${SELECTED_BUS_PORT}" == "EXIT" ]; then
+			exit 0
+		fi
+		# treate selected number
```
Comment (jeongchanKim) [inline os/dbuild.sh]: treate -> treat

File: os/dbuild.sh [hunk 5]
```diff
@@ -247,6 +250,103 @@ function SELECT_CONFIG()
 	CONFIGURE ${BOARD}/${CONFIG} || exit 1
 }
 
+function SELECT_PORT()
```
Comment (jeongchanKim) [inline os/dbuild.sh]: this function calls SELECT_DL. 
how about moving this function below the SELECT_DL?

**Comment by seinfra** [comment]
Target : [e48ecc7bab59884734387a11d79447cde7012318] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [e48ecc7bab59884734387a11d79447cde7012318] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9c98c0301675add068234865b1e2893c0a78a789] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [9c98c0301675add068234865b1e2893c0a78a789] - Code Rule Check (C++) OK.

File: os/dbuild.sh [hunk 6]
```diff
@@ -284,6 +290,100 @@ function SELECT_DL
...
+					PORT_INFO=`lsusb -t | sed -n -e ''${LINE_INFO[${LINE_IDX}*2 - 1]}','${LINE_INFO[${LINE_IDX}*2 + 1]}'p' -e 's/.Port//' | awk '!/Bus/{print $3}' | sed 's/://'`
+					PORT_LIST=`echo ${PORT_INFO} | xargs -n1 | sort -u | xargs`
+					BUS_PORT_INFO[${BUS_NUM}]=${PORT_LIST}
+					break
+				fi
+				((LINE_IDX=LINE_IDX+1))
+			done
+		done
+
+		# Print Bus and Port inforation
```
Comment (JeongJunSik) [inline os/dbuild.sh]: information

**Comment by seinfra** [comment]
Target : [9c98c0301675add068234865b1e2893c0a78a789] - Code Rule Check OK.

**Comment by seinfra** [comment]
Target : [9c98c0301675add068234865b1e2893c0a78a789] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [9c98c0301675add068234865b1e2893c0a78a789] - Code Rule Check (C++) OK.

**Pull request closed without merging**